github.go

```diff
@@ -144,6 +116,31 @@ func (base GithubImporter) GetToken(i Import) (wharfapi.Token, error) {
 	return token, err
 }
 
+func (base GithubImporter) GetProvider(i Import) (wharfapi.Provider, error) {
+	var provider wharfapi.Provider
+	var err error
+
+	if i.ProviderId != 0 {
+		provider, err = base.WharfClient.GetProviderById(i.ProviderId)
+		if err != nil {
+			return provider, err
+		} else if provider.ProviderID == 0 {
+			err = fmt.Errorf("provider with id %v not found", i.ProviderId)
+		} else if provider.URL != i.Url {
+			err = fmt.Errorf("invalid url in provider %v", provider.URL)
+		} else if provider.UploadURL != i.UploadUrl {
+			err = fmt.Errorf("invalid upload url in provider %v", provider.UploadURL)
+		}
+	} else {
+		provider, err = base.WharfClient.GetProvider("github", i.Url, i.UploadUrl, base.Token.TokenID)
```

Comment:
I see that this code was like this previously as well, but could you change this `base.Token` to be a parameter to this method instead and use the return value from the `GetToken` method and feed it into this method

Goes hand in hand with: [G31: Hidden Temporal Couplings](https://moderatemisbehaviour.github.io/clean-code-smells-and-heuristics/general/g31-hidden-temporal-couplings.html)

Reply:
Done now imported as `token wharfapi.Token`.